packages/medusa/src/scripts/migrate-to-pricing-module.ts

```diff
@@ -0,0 +1,232 @@
+import { CurrencyService, PriceListService } from "../services"
+import { IPricingModuleService, PricingTypes } from "@medusajs/types"
+
+import { AwilixContainer } from "awilix"
+import { PriceList } from "../models"
+import { createDefaultRuleTypes } from "./utils/create-default-rule-types"
+import dotenv from "dotenv"
+import express from "express"
+import loaders from "../loaders"
+import { migrateProductVariantPricing } from "./utils/migrate-money-amounts-to-pricing-module"
+
+dotenv.config()
+
+const BATCH_SIZE = 10
+
+const migratePriceLists = async (container: AwilixContainer) => {
+  const pricingModuleService: IPricingModuleService = container.resolve(
+    "pricingModuleService"
+  )
+  let offset = 0
+
+  const priceListCoreService: PriceListService =
+    container.resolve("priceListService")
+
+  const remoteQuery = container.resolve("remoteQuery")
+
+  const [_, totalCount] = await priceListCoreService.listAndCount({})
+
+  while (offset < totalCount) {
+    const corePriceLists = await priceListCoreService.list(
+      {},
+      {
+        take: BATCH_SIZE,
+        skip: offset,
+        relations: ["customer_groups", "prices", "prices.variants"],
+      }
+    )
+
+    const pricingModulePriceLists = await pricingModuleService.listPriceLists(
+      { id: corePriceLists.map(({ id }) => id) },
+      {
+        take: BATCH_SIZE,
+        skip: offset,
+        select: ["id"],
+      }
+    )
+
+    const priceListIdsToUpdateSet = new Set<string>(
+      pricingModulePriceLists.map(({ id }) => id)
+    )
+
+    const priceListsToCreate: PriceList[] = corePriceLists.filter(
+      ({ id }) => !priceListIdsToUpdateSet.has(id)
+    )
+
+    const priceListsToUpdate: PriceList[] = corePriceLists.filter(({ id }) =>
+      priceListIdsToUpdateSet.has(id)
+    )
```

Comment:
suggestion: create the two arrays before and then loop only once over the corePriceLists and push in one or the other depending on the id constraint